include/ensmallen_bits/nsga2/nsga2.hpp

```diff
@@ -51,7 +51,12 @@ namespace ens {
  * see the documentation on function types included with this distribution or
  * on the ensmallen website.
  */
-class NSGA2
+template<
+  typename MatType,
+  typename ColType = typename ForwardType<MatType>::bvec,
+  typename CubeType = typename ForwardType<MatType>::bcube
+>
+class NSGA2Type
```

Comment:
We should also update the NSGA2 documentation to indicate what's necessary to use NSGA2 with non-`arma::mat` types.